mm2src/coins/z_coin.rs

```diff
@@ -649,98 +465,128 @@ impl<'a> UtxoCoinWithIguanaPrivKeyBuilder for ZCoinBuilder<'a> {
     async fn build(self) -> MmResult<Self::ResultCoin, Self::Error> {
         let utxo = self.build_utxo_fields_with_iguana_priv_key(self.priv_key()).await?;
         let utxo_arc = UtxoArc::new(utxo);
-        let db_name = format!("{}_CACHE.db", self.ticker);
-        let mut db_dir_path = self.db_dir_path;
-
-        db_dir_path.push(&db_name);
-        let sqlite = tokio::task::block_in_place(move || {
-            if !db_dir_path.exists() {
-                let default_cache_path = PathBuf::from(format!("./{}", db_name));
-                if !default_cache_path.exists() {
-                    return MmError::err(ZCoinBuildError::SaplingCacheDbDoesNotExist {
-                        path: std::env::current_dir()?.join(&default_cache_path).display().to_string(),
-                    });
-                }
-                std::fs::copy(default_cache_path, &db_dir_path)?;
-            }
-
-            let sqlite = Connection::open(db_dir_path)?;
-            init_db(&sqlite)?;
-            Ok(sqlite)
-        })?;
 
         let (_, my_z_addr) = self
             .z_spending_key
             .default_address()
             .map_err(|_| MmError::new(ZCoinBuildError::GetAddressError))?;
 
-        let dex_fee_addr = decode_payment_address(z_mainnet_constants::HRP_SAPLING_PAYMENT_ADDRESS, DEX_FEE_Z_ADDR)
+        let dex_fee_addr = decode_payment_address(self.consensus_params.hrp_sapling_payment_address(), DEX_FEE_Z_ADDR)
             .expect("DEX_FEE_Z_ADDR is a valid z-address")
             .expect("DEX_FEE_Z_ADDR is a valid z-address");
 
-        let z_tx_prover = tokio::task::block_in_place(LocalTxProver::with_default_location)
+        let z_tx_prover = async_blocking(LocalTxProver::with_default_location)
+            .await
             .or_mm_err(|| ZCoinBuildError::ZCashParamsNotFound)?;
 
-        let my_z_addr_encoded = encode_payment_address(z_mainnet_constants::HRP_SAPLING_PAYMENT_ADDRESS, &my_z_addr);
-        let my_z_key_encoded = encode_extended_spending_key(
-            z_mainnet_constants::HRP_SAPLING_EXTENDED_SPENDING_KEY,
-            &self.z_spending_key,
-        );
+        let my_z_addr_encoded = encode_payment_address(self.consensus_params.hrp_sapling_payment_address(), &my_z_addr);
+
+        let evk = ExtendedFullViewingKey::from(&self.z_spending_key);
+        let (sync_state_connector, light_wallet_db) = match &self.z_coin_params.mode {
+            ZcoinRpcMode::Native => {
+                return MmError::err(ZCoinBuildError::NativeModeIsNotSupportedYet);
```

Comment:
We can support both of them, but I decided to use the same database architecture for both of these modes. It will take a bit of time to refactor for the native mode, which I will do on the next iteration.